python/ray/actor.py

```diff
@@ -612,22 +681,57 @@ def _actor_method_call(self,
 
         args = args or []
         kwargs = kwargs or {}
-        function_signature = self._ray_method_signatures[method_name]
-
-        if not args and not kwargs and not function_signature:
-            list_args = []
+        if self._ray_is_cross_language:
+            if not worker.load_code_from_local:
+                raise Exception("Cross language feature needs "
+                                "--load-code-from-local to be set.")
+            if self._ray_actor_language == Language.PYTHON:
+                # To compatible with ray.signature.recover_args
+                function_signature = signature.ANY_FUNCTION_SIGNATURE
+                list_args = signature.flatten_args(function_signature, args,
+                                                   kwargs)
+            else:
+                if kwargs:
+                    raise Exception("Cross language remote actor method "
+                                    "not support kwargs.")
+                list_args = args
+            if self._ray_actor_language == Language.PYTHON:
+                function_descriptor = PythonFunctionDescriptor(
+                    self._ray_actor_creation_function_descriptor.module_name,
+                    method_name,
+                    self._ray_actor_creation_function_descriptor.class_name)
+            elif self._ray_actor_language == Language.JAVA:
+                function_descriptor = JavaFunctionDescriptor(
+                    self._ray_actor_creation_function_descriptor.class_name,
+                    method_name,
+                    # Currently not support call actor method with signature.
+                    "")
+            else:
+                raise NotImplementedError("Cross language remote actor method "
+                                          "not support language {}".format(
+                                              self._ray_actor_language))
```

Comment:
How about put the code to deal with specific languages into a separate file? So later if we add cross-lang capabilities for other languages, we don't need to change code all over the places.